core/consensus/babe/src/lib.rs

```diff
@@ -85,6 +85,9 @@ mod aux_schema;
 mod tests;
 pub use babe_primitives::AuthorityId;
 
+/// Maximum uncles generations we may provide to the runtime.
+const MAX_UNCLE_GENERATIONS: u32 = 8;
```

Comment:
this should be drawn from the runtime instead

Reply:
Runtime calls are expensive. And this will be called for each block verification. My reasoning is that it is better to pick a hard limit, pass a few extra headers and let the runtime sort it out later rather than make another call into WASM.

Reply:
it will be called for every block creation only. but yeah, this is fine for now.

Reply:
From what I see `InherentData` is populated for verification as well.